src/rdb_protocol/datum_stream.cc

```diff
@@ -1991,11 +1991,13 @@ std::vector<changespec_t> union_datum_stream_t::get_changespecs() {
 range_datum_stream_t::range_datum_stream_t(bool _is_infinite_range,
                                            int64_t _start,
                                            int64_t _stop,
+                                           int64_t _step,
                                            backtrace_id_t bt)
     : eager_datum_stream_t(bt),
       is_infinite_range(_is_infinite_range),
```

Comment:
I'm throwing a logic error [here](https://github.com/rethinkdb/rethinkdb/pull/5545/files#diff-c13480669499466d2358b2c0cd04c327R798) when `step == 0`, would an infinite stream be preferable?